src/main/java/mekanism/common/integration/framedblocks/FramedBlocksIntegration.java

```diff
@@ -0,0 +1,83 @@
+package mekanism.common.integration.framedblocks;
+
+import mekanism.api.MekanismAPI;
+import mekanism.api.chemical.Chemical;
+import mekanism.common.Mekanism;
+import net.minecraft.core.particles.ParticleType;
+import net.minecraft.core.registries.Registries;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.TagKey;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.fml.loading.FMLEnvironment;
+import net.neoforged.neoforge.client.event.ModelEvent;
+import net.neoforged.neoforge.client.event.RegisterParticleProvidersEvent;
+import net.neoforged.neoforge.fluids.FluidType;
+import net.neoforged.neoforge.registries.DeferredHolder;
+import net.neoforged.neoforge.registries.DeferredRegister;
+import xfacthd.framedblocks.api.camo.CamoContainerFactory;
+import xfacthd.framedblocks.api.util.FramedConstants;
+
+public final class FramedBlocksIntegration {
+
+    private static final DeferredRegister<CamoContainerFactory<?>> CAMO_FACTORIES = DeferredRegister.create(
+            FramedConstants.CAMO_CONTAINER_FACTORY_REGISTRY_KEY,
+            Mekanism.MODID
+    );
+    private static final DeferredRegister<ParticleType<?>> PARTICLE_TYPES = DeferredRegister.create(
+            Registries.PARTICLE_TYPE, Mekanism.MODID
+    );
+
+    static final DeferredHolder<CamoContainerFactory<?>, ChemicalCamoContainerFactory> CHEMICAL_FACTORY =
+            CAMO_FACTORIES.register("chemical", ChemicalCamoContainerFactory::new);
+    static final DeferredHolder<ParticleType<?>, ChemicalParticleOptions.Type> CHEMICAL_PARTICLE =
+            PARTICLE_TYPES.register("chemical", ChemicalParticleOptions.Type::new);
+
+    public static void init(IEventBus modBus) {
+        CAMO_FACTORIES.register(modBus);
+        PARTICLE_TYPES.register(modBus);
+
+        if (FMLEnvironment.dist.isClient()) {
+            ClientEvents.init(modBus);
+        }
+    }
+
+    public static final class Constants {
+
+        /**
+         * Chemicals tagged with this tag cannot be inserted into a framed blocks
+         */
+        public static final TagKey<Chemical> CHEMICAL_BLACKLISTED = TagKey.create(
```

Comment:
We should probably add this in `MekanismLangProvider#addTags` or EMI will complain in dev that it doesn't have a translation.

Also do you think it makes sense to expose this constant in `MekanismAPITags`?